app/org/thp/cortex/services/DockerJobRunnerSrv.scala

```diff
@@ -50,89 +51,115 @@ class DockerJobRunnerSrv(
-        .readOnly(false)
-        .build()
-    )
-    val cacertsFile = jobDirectory.resolve("input").resolve("cacerts")
-    val containerConfigBuilder = ContainerConfig
-      .builder()
-      .hostConfig(hostConfigBuilder.build())
-      .image(dockerImage)
-      .cmd("/job")
+  private def generateErrorOutput(containerId: String, f: Path) = {
+    logger.warn(s"the runner didn't generate any output file $f")
+    for {
+      output <- javaClient.getLogs(containerId)
+      report = Json.obj("success" -> false, "errorMessage" -> output)
+      _ <- Try(Files.write(f, report.toString.getBytes(StandardCharsets.UTF_8)))
+    } yield report
+  }
 
-    val containerConfig =
-      if (Files.exists(cacertsFile)) containerConfigBuilder.env(s"REQUESTS_CA_BUNDLE=/job/input/cacerts").build()
-      else containerConfigBuilder.build()
-    val containerCreation = client.createContainer(containerConfig)
-    //          Option(containerCreation.warnings()).flatMap(_.asScala).foreach(logger.warn)
+  def run(jobDirectory: Path, dockerImage: String, timeout: Option[FiniteDuration])(implicit executionContext: ExecutionContext): Try[Unit] = {
+    val to = timeout.getOrElse(FiniteDuration(5000, TimeUnit.SECONDS))
 
-    logger.debug(s"Container configuration: $containerConfig")
-    logger.info(
-      s"Execute container ${containerCreation.id()}\n" +
-        s"  timeout: ${timeout.fold("none")(_.toString)}\n" +
-        s"  image  : $dockerImage\n" +
-        s"  volume : ${jobDirectory.toAbsolutePath}:/job" +
-        Option(containerConfig.env()).fold("")(_.asScala.map("\n  env    : " + _).mkString)
-    )
+    if (autoUpdate) Try(javaClient.pullImage(dockerImage, to))
 
-    val timeoutSched = timeout.map(to =>
-      system.scheduler.scheduleOnce(to) {
-        logger.info("Timeout reached, stopping the container")
-        client.removeContainer(containerCreation.id(), DockerClient.RemoveContainerParam.forceKill())
-      }
-    )
-    val execution = Try {
-      client.startContainer(containerCreation.id())
-      client.waitContainer(containerCreation.id())
-      ()
-    }
-    timeoutSched.foreach(_.cancel())
-    val outputFile = jobDirectory.resolve("output").resolve("output.json")
-    if (!Files.exists(outputFile) || Files.size(outputFile) == 0) {
-      logger.warn(s"The worker didn't generate output file.")
-      val output = Try(client.logs(containerCreation.id(), LogsParam.stdout(), LogsParam.stderr()).readFully())
-        .fold(e => s"Container logs can't be read (${e.getMessage})", identity)
-      val message = execution.fold(e => s"Docker creation error: ${e.getMessage}\n$output", _ => output)
+    for {
+      containerId <- javaClient.prepare(dockerImage, jobDirectory, jobBaseDirectory, dockerJobBaseDirectory, to)
+      timeoutScheduled = timeout.map(to =>
+        system.scheduler.scheduleOnce(to) {
+          logger.info("Timeout reached, stopping the container")
+          javaClient.clean(containerId)
```

Comment:
When the timeout is reached, you should not only clean the container, you should kill it.